File: backend/ShelfContext.Contract/Queries/GetLibrarySummary/LibrarySummary.cs
using System.Text.Json.Serialization;

namespace ShelfContext.Contract.Queries.GetLibrarySummary
{
    public record LibrarySummary(
        [property: JsonPropertyName("user_id")] Guid UserId,
        [property: JsonPropertyName("book_count")] int BookCount,
        [property: JsonPropertyName("shelves")] ICollection<ShelfSummary> Shelves,
        [property: JsonPropertyName("tags")] ICollection<TagSummary> Tags)
    {
    }
}

File: onlib/src/app/business/models/shelves/librarySummary.ts
import { ShelfSummary } from './shelfSummary';
import { TagSummary } from './tagSummary';

export interface LibrarySummary {
  user_id: string;
  shelves: ShelfSummary[];
  tags: TagSummary[];
  book_count: number;
}
